spog/api/src/endpoints/dashboard.rs

```diff
@@ -1,26 +1,75 @@
 use crate::app_state::AppState;
 use crate::search;
 use crate::service::v11y::V11yService;
-use actix_web::web::ServiceConfig;
+use actix_web::web::{PayloadConfig, ServiceConfig};
 use actix_web::{web, HttpResponse};
 use actix_web_httpauth::extractors::bearer::BearerAuth;
-use spog_model::dashboard::{CSAFStatus, CveStatus, DashboardStatus, SbomStatus};
+use spog_model::dashboard::{CSAFStatus, CveStatus, DashboardStatus, SbomStatus, UserPreferences};
 use std::sync::Arc;
 use tracing::instrument;
 use trustification_api::search::SearchOptions;
 use trustification_auth::authenticator::Authenticator;
 use trustification_infrastructure::new_auth;
 
-pub(crate) fn configure(auth: Option<Arc<Authenticator>>) -> impl FnOnce(&mut ServiceConfig) {
-    |config: &mut ServiceConfig| {
+pub(crate) fn configure(auth: Option<Arc<Authenticator>>, payload_limit: usize) -> impl FnOnce(&mut ServiceConfig) {
+    move |config: &mut ServiceConfig| {
         config.service(
             web::scope("/api/v1/dashboard")
                 .wrap(new_auth!(auth))
-                .service(web::resource("/status").to(get_status)),
+                .service(web::resource("/status").to(get_status))
+                .service(
+                    // web::scope("/api/v1/dashboard")
+                    web::resource("/userPreferences/{id}").to(user_preferences_receive),
```

Comment:
Sorry guys for jumping into the discussion directly but I've read about the `/api/v1/dashboard/userPreferences/{USERNAME}` endpoint.
If I'm not misunderstanding it or missing something, I strongly suggest you to avoid having the username as a parameter but instead use the username actually logged to retrieve his/her preferences (if possible).
Basically an authenticated call to the `/api/v1/dashboard/userPreferences/` endpoint should be enough to retrieve the user doing the authenticated call itself.
Otherwise I'm concerned about having to guarantee no information can be leaked between users.

Reply:
I think the same should be true for the other endpoints.